### PR TITLE
Containerize the simulator

### PR DESCRIPTION
**Pull request by aswanson-nr**
## Summary

[NOTE]: # ( Provide a brief overview of the changes. )
* Add a Dockerfile
* Add a Github action to publish the docker image
* ~~Allow passing of a WEB_DRIVER_URL environment variable, to tell the
selenium web driver where to run. This is done because chrome cannot be
executed in the simulator's docker container.~~ Replaced with the standard SELENIUM_REMOTE_URL

## Checklist
[NOTE]: # ( Just check the ones applicable to this pull request. )
- [x] Documentation updated

## Deployment Configuration for Testing
[NOTE]: # ( Provide a deployment configuration to use during manual testing, if applicable. )

## Related Issues
[NOTE]: # ( Provide links to any related Github issues. )

## Related Pull Requests
[NOTE]: # ( Provide links to any related pull requests. )

## Reviewers
[NOTE]: # ( Mention reviewers here! )

### CODE REVIEW

**Comment by aswanson-nr** [comment]
Not ready to be merged yet, still working on the documentation.

**Comment by aswanson-nr** [comment]
I discovered that the url for the standalone selenium driver can be passed in as `SELENIUM_REMOTE_URL`, thus removing the need to handle passing that value around the application. I am going to remove the code related to handling the `WEB_DRIVER_URL` since the preexisting variable already takes care it.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM node:lts
+
+RUN apt-get clean all
+RUN apt-get update
+RUN apt update
```
Comment (nr-kkenney) [inline Dockerfile:5]: i ran hadolint on this and got a couple items of feedback.

- `DL3009 info: Delete the apt-get lists after installing something`
- `DL3027 warning: Do not use apt as it is meant to be a end-user tool, use apt-get or apt-cache instead`

i think if you update this to:
```sh
RUN apt-get update && \
    apt-get clean && \
    rm -rf /var/lib/apt/lists/*
```

should be good

Comment (aswanson-nr) [reply Dockerfile]: Done!

**Review by nr-kkenney — APPROVED** [review]
lgtm

**Pull request merged**